### PR TITLE
feat: ✨ adds StatusChangeReviewCards

### PR DESCRIPTION
**Pull request by fpigeonjr**
## What changed

- handles status change requests
- adds StatusChangeReviewCard
- adds decimal place to formatter function
- moves mocks module to tests directory

## Issue

#2307 #2441

## How to test

1. goto Agreement 1/9
2. submit a BLI for approval
3. goto Agreement Change Requests
4. see StatusChangeReviewCard

## Screenshots

https://github.com/HHS/OPRE-OPS/assets/4629398/bac100a8-e9a6-4be1-9d2c-8000e3aa0783

## Definition of Done Checklist
- [X] OESA: Code refactored for clarity
- [X] OESA: Dependency rules followed
- [X] Automated unit tests updated and passed
- [X] Automated integration tests updated and passed
- [X] Automated quality tests updated and passed
- [X] Automated load tests updated and passed
- [X] Automated a11y tests updated and passed
- [X] Automated security tests updated and passed
- [X] 90%+ Code coverage achieved
- [-] Form validations updated

### CODE REVIEW

File: frontend/cypress/e2e/reviewAgreement.cy.js [hunk 1]
```diff
@@ -343,65 +343,77 @@ describe("agreement review workflow", () => {
             // get table rows and should have text In Review
             cy.get("@table-rows").eq(0).should("contain", "In Review");
             cy.get("@table-rows").eq(1).should("contain", "In Review");
-
             //approve agreement
-            cy.visit("/agreements?filter=for-approval").wait(1000);
-            cy.get("tbody").children().as("table-rows").should("exist");
+            cy.visit("/agreements?filter=change-requests").wait(1000);
```
Comment (fpigeonjr) [inline frontend/cypress/e2e/reviewAgreement.cy.js:347]: we don't have a way to approve change requests so commented out for now

File: frontend/src/pages/agreements/review/ReviewAgreement.jsx [hunk 2]
```diff
@@ -124,34 +139,41 @@ export const ReviewAgreement = () => {
...
-                workflow_action: workflowAction
-            })
-                .unwrap()
-                .then((fulfilled) => {
-                    console.log("BLI Status Updated:", fulfilled);
+
+            let promises = selectedBLIsWithStatusAndNotes.map((budgetLine) => {
+                const { id, data: cleanExistingBLI } = cleanBudgetLineItemForApi(budgetLine);
+                return updateBudgetLineItem({ id, data: cleanExistingBLI })
+                    .unwrap()
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/review/ReviewAgreement.jsx:148]: crux of the PR here: use patch agreement to generate a change request

File: frontend/src/components/ChangeRequests/StatusChangeReviewCard/StatusChangeReviewCard.test.jsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { render, screen } from "@testing-library/react";
+import StatusChangeReviewCard from "./StatusChangeReviewCard";
+
+describe("StatusChangeReviewCard", () => {
+    it.todo("should render the StatusChangeReviewCard component", () => {
+        render(
```
Comment (fpigeonjr) [inline frontend/src/components/ChangeRequests/StatusChangeReviewCard/StatusChangeReviewCard.test.jsx:6]: will flesh this out more tomorrow

**Review by johndeange — APPROVED** [review]

File: frontend/src/pages/agreements/review/ReviewAgreement.jsx [hunk 4]
```diff
@@ -109,9 +108,29 @@ export const ReviewAgreement = () => {
...
-            const bli_ids = getSelectedBudgetLines(budgetLines).map((bli) => bli.id);
-            const user_id = activeUser?.id;
+            const selectedBudgetLines = getSelectedBudgetLines(budgetLines);
+            let selectedBLIsWithStatusAndNotes = [];
+
+            switch (action) {
+                case actionOptions.CHANGE_DRAFT_TO_PLANNED:
+                    selectedBLIsWithStatusAndNotes = selectedBudgetLines.map((bli) => {
+                        return { id: bli.id, status: BLI_STATUS.PLANNED, requestor_notes: notes };
+                    });
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/review/ReviewAgreement.jsx:118]: paired down to only `id, status, and notes`
- thx @stevtek for the help with this

**Review by stevtek — APPROVED** [review]

**Pull request merged**